File: Permutacja.java
import java.util.Arrays;

public class Permutacja {
    protected static void swap(int[] a, int i, int j) {
        int t = a[i];
        a[i] = a[j];
        a[j] = t;
    }

    protected static void reverse(int[] p, int m) {
        int i = 1;
        int j = m;
        while (i < j) {
            swap(p,i,j);
            i++;
            j--;
        }
    }

    protected static void antylex( int[] p, int m ) {
        if( m == 1) {
            //System.out.println(Arrays.toString(p));
        } else {
            for(int i=1; i <= m; i++ ) {
                antylex(p, m-1);
                if( i < m ) {
                    swap( p, i,m );
                    reverse(p, m-1);
                }
            }
        }
    }

    public static void main(String[] args) {
        int n = Integer.parseInt(args[0]);

        int[] p = new int[n+1];
        for( int i=1; i <= n; i++ ) {
            p[i] = i;
        }

        //System.out.println(Arrays.toString(p));

        antylex(p, n);
    }
}

File: permutacja.py
def reverse(p,m):
    i = 1
    j = m
    while i < j:
        p[i], p[j] = p[j], p[i]
        i += 1
        j -= 1

def antylex( p,m ):
    if m == 1:
        m
        #print p
    else:
        for i in range( 1, m+1):
            antylex( p, m-1)
            if i < m:
                p[i], p[m] = p[m], p[i]
                reverse( p, m-1)
            i+=1


n = 11;
p = [0];
for i in range(1,n+1):
    p.append(i)

#print p
antylex(p, n )
